project/dexter-monitor/public/ctrl/currentUserCtrl.js

```diff
@@ -25,46 +25,49 @@
-        {field:'department',        displayName:'Group',    width: 284,     cellClass: 'grid-align',    headerTooltip: 'Group'},
-        {field:'title',             displayName:'Title',    width: 194,     cellClass: 'grid-align',    headerTooltip: 'Title'},
-        {field:'employeeNumber',    displayName:'Number',   width: 184,     cellClass: 'grid-align',    headerTooltip: 'Number'}
+        {field:'projectName',       displayName:'Project',  width: '20%',   cellClass: 'grid-align',    headerTooltip: 'Project name'},
+        {field:'userId',            displayName:'ID',       width: '15%',   headerTooltip: 'User ID',
+            cellTemplate: '<div class="ui-grid-cell-contents grid-align" uib-tooltip="Click if you want to see extra information of \'{{COL_FIELD}}\'"' +
+            ' tooltip-placement="top" tooltip-append-to-body="true"><a href ng-click="grid.appScope.getExtraInfo(row.entity)">{{COL_FIELD}}</a></div>'},
+        {field:'name',              displayName:'Name',     width: '15%',   cellClass: 'grid-align',    headerTooltip: 'User name'},
+        {field:'department',        displayName:'Group',    width: '20%',   cellClass: 'grid-align',    headerTooltip: 'Group name'},
+        {field:'title',             displayName:'Title',    width: '15%',   cellClass: 'grid-align',    headerTooltip: 'User title'},
+        {field:'employeeNumber',    displayName:'Number',   width: '15%',   cellClass: 'grid-align',    headerTooltip: 'User number'}
     ];
 
     initialize();
 
     function initialize() {
-        user.gridOptions = createGrid(columnDefs);
+        $scope.gridOptions = createGrid(columnDefs);
         loadUserList();
-        setGridExportingFileNames(user.gridOptions, USER_FILENAME_PREFIX);
+        $scope.time = new Date().toLocaleString();
+        setGridExportingFileNames($scope.gridOptions, USER_FILENAME_PREFIX + '-' + $scope.time);
     }
 
     function loadUserList() {
-        $http.get('/api/v2/user')
-            .then((res) => {
-                if (!isHttpResultOK(res)) {
-                    $log.error('Failed to load user list');
-                    return;
-                }
-
-                user.gridOptions.data = res.data.rows;
+        UserService.getUserList()
+            .then((rows) => {
+                $scope.gridOptions.data = rows;
             })
             .catch((err) => {
                 $log.error(err);
             });
     }
 
-    user.getExtraInfo = function() {
-        const userIdList = _.map(user.gridOptions.data, 'userId');
-        UserService.getExtraInfoByUserIdList(userIdList)
-            .then((rows) => {
-                if(rows) {
-                    user.gridOptions.data = _.sortBy(rows, 'userId');
+    $scope.getExtraInfo = function(entity) {
+        if (entity.name && entity.department && entity.title && entity.employeeNumber)
+            return;
+
+        UserService.getExtraInfoByUserId(entity.userId)
```

Comment:
If UserService.getExtrainfoByUserId() uses DB or REST Service, it is not good for performance.
Why don't you use one service for extra user list?